File: hackerrank/problem solving/algorithms/1 - warmup/5 - a very big sum/with morfo library/a_very_big_sum.rs
// https://www.hackerrank.com/challenges/a-very-big-sum/problem?isFullScreen=true

fn main() {
    morfo::skip_input_ln();
    let arr: Vec<i64> = morfo::read_ln();
    println!("{}", a_very_big_sum(&arr));
}

// n: length of arr
// T: O(n)
// S: O(1) extra space
fn a_very_big_sum(arr: &[i64]) -> i64 {
    arr.iter().sum()
}

File: hackerrank/problem solving/algorithms/1 - warmup/5 - a very big sum/with morfo library/a_very_big_sum.go
// https://www.hackerrank.com/challenges/a-very-big-sum/problem?isFullScreen=true

package main

import (
    "fmt"
    "morfo/morfoio"
    "morfo/morfomath"
)

func main() {
    morfoio.SkipInputLn()
    arr := morfoio.ReadLn[int64]()
    fmt.Println(aVeryBigSum(arr))
}

// n: length of arr
// T: O(n)
// S: O(1) extra space
func aVeryBigSum(arr []int64) int64 {
    return morfomath.Sum(arr)
}
